### ISSUE TITLE
drag and drop broken

### ISSUE THREAD

**Issue by Simon-Laux** [flathub/chat.delta.desktop#137]
In some circumstances drag and drop for sending files with deltachat is not working in the flatpak.

It works in KDE on the Steam Deck (steam os 3.5.19; kde plasma 5.27.5; graphics X11), but for it does not work in gnome (probably wayland).
This was also reported by other users, but I don't remember where.

We need more information to figgure out what went wrong.

**Comment by muelli** [comment]
yeah, I guess that the sandbox restricts access to the dropped file path.
I think the proper is to support dnd portal https://docs.flatpak.org/en/latest/portal-api-reference.html#gdbus-org.freedesktop.portal.FileTransfer

but Electron is behind: https://github.com/electron/electron/issues/30650

one can probably work around this problem by opening the sandbox. I would try to avoid that, though.

**Comment by Simon-Laux** [comment]
I just gave it full file access and it worked, so that confirms your guess.

maybe we should already use the portal api without waiting for an electron implementation. There are some wrapper libraries around that should make it easier than using dbus directly: 
- rust: https://bilelmoussaoui.github.io/ashpd/ashpd/documents/struct.FileTransfer.html
- c: https://github.com/flatpak/libportal/

I think using the rust lib would be easier, like wrap the functions needed with https://napi.rs and then use them from nodejs. Or maybe even just make a small cli that is called by the dc backend when it is run in flatpak, whatever is easiest.
